lib/helpers/getBasicDomNodeProfile.js

```diff
@@ -13,30 +13,43 @@ function getAttr(el, name) {
   return null;
 }
 
-export default function getBasicDomNodeProfile(el) {
-  let allNodeAttributes = getDomNodeAttributes(el, EXCLUDE_VALUE_REGEX);
-
-  /**
-   * This is a short, unique CSS selector for quickly grabbing this node from the DOM.
-   * It serves a different purpose than a full node 'path', which potentially holds
-   * much more information about the DOM structure.
-   *
-   * NOTE(dabrady) We are putting this in the `all_attrs` blob out of a variety of
-   * concerns, but primarily because it will let us evalutate our use-case for this
-   * data faster and with minimal effort.
-   */
-  const uniqueSelector = generateUniqueSelector(el);
-  if (uniqueSelector) {
-    // Don't include uniqueSelector if it doesn't have a value.
-    allNodeAttributes = { ...allNodeAttributes, unique_selector: uniqueSelector };
+/**
+ * This is a short, unique CSS selector for quickly grabbing this node from the DOM.
+ * It serves a different purpose than a full node 'path', which potentially holds
+ * much more information about the DOM structure.
+ */
+function generateUniqueSelector(el) {
+  // NOTE(dabrady) The `unique-selector` library assumes the given node is part of a DOM, and will throw errors
+  // if that's not true. So guarding against that here.
+  if (document.contains(el)) {
+    try {
+      return uniqueSelector(el);
+    } catch (TypeError) {
+      // NOTE(dabrady) If the target element is not part of the DOM by the time `generateUniqueSelector`
+      // tries to traverse the element's parent tree, it will throw a null-pointer exception. This try-catch
+      // prevents those errors from bubbling up to our clients' applications.
+      return null;
+    }
   }
+  return null;
+}
 
+export default function getBasicDomNodeProfile(el) {
   return {
     class: getAttr(el, 'class') || null,
     href: (typeof el.href === 'object' ? null : el.href) || null,
     id: getAttr(el, 'id') || null,
     name: getAttr(el, 'name') || null,
-    all_attrs: allNodeAttributes,
+    all_attrs: {
+      ...getDomNodeAttributes(el, EXCLUDE_VALUE_REGEX),
+      /**
+       * NOTE(dabrady) We are putting this in the `all_attrs` blob out of a variety of
+       * concerns, but primarily because it will let us evalutate our use-case for this
+       * data faster and with minimal effort.
+       */
+      // Don't include uniqueSelector if it couldn't be calculated.
+      unique_selector: generateUniqueSelector(el) || undefined,
```

Comment:
As part of this, I simplified the hotfix I made earlier around ensuring the `unique_selector` field is not added to `all_attrs` unless it has a value.

When a JS object is serialized into JSON and sent over the wire, `undefined` properties are ignored. You can easily confirm this by smoke testing the JSON serializer directly in a console:
```javascript
> JSON.stringify({ a: 42, b: null, c: undefined, d: false })
< "{\"a\":42,\"b\":null,\"d\":false}"
```

Reply:
This is very clever.